src/api/middlewares/url_resolver_middleware.py

```diff
@@ -0,0 +1,15 @@
+from django.http import HttpResponse
+from django.urls import resolve, Resolver404
+
+class URLResolverMiddleware:
+    def __init__(self, get_response):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        try:
+            resolve(request.path)
+        except Resolver404:
+            return HttpResponse("This endpoint does not exist.", status=404)
+        
+        response = self.get_response(request)
```

Comment:
You can just return it without setting the variable `response = self.get_response(request)`

Reply:
ok